src/video/wayland/SDL_waylanddatamanager.c

```diff
@@ -136,29 +137,20 @@ static ssize_t read_pipe(int fd, void **buffer, size_t *total_length, SDL_bool n
     return bytes_read;
 }
 
-#define MIME_LIST_SIZE 4
-
-static const char *mime_conversion_list[MIME_LIST_SIZE][2] = {
-    { "text/plain", TEXT_MIME },
-    { "TEXT", TEXT_MIME },
-    { "UTF8_STRING", TEXT_MIME },
-    { "STRING", TEXT_MIME }
-};
-
-const char *Wayland_convert_mime_type(const char *mime_type)
+static size_t
+Wayland_send_data(void *data, size_t length, int fd)
 {
-    const char *found = mime_type;
+    size_t result = 0;
 
-    size_t index = 0;
-
-    for (index = 0; index < MIME_LIST_SIZE; ++index) {
-        if (SDL_strcmp(mime_conversion_list[index][0], mime_type) == 0) {
-            found = mime_conversion_list[index][1];
-            break;
+    if (length > 0 && data != NULL) {
+        while (write_pipe(fd, data, length, &result) > 0) {
+            // Just keep spinning
         }
+        close(fd);
+        SDL_free(data);
```

Comment:
Is this freeing data received from the callback?

Reply:
Yes

Reply:
I would avoid doing that and just let the application deal with freeing it when it needs to. Otherwise the application would have to create a copy of the data every time a paste operation is needed.

Reply:
Yes... it does put a bit more demand on the application implementation though. But I also think it's a more sound approach. Just awkward having to retain and free returned data. I'll fix